src/ConfCrypt/Commands.hs

```diff
@@ -52,14 +53,26 @@ class Monad m => Command a m where
     evaluate :: a -> m [T.Text]
 
 -- | Read and return the full contents of an encrypted file. Provides support for using a local RSA key or an externl KMS service
-data ReadConfCrypt = ReadConfCrypt
+data ReadConfCrypt = ReadConfCrypt {rTemplate :: Maybe T.Text}
+    deriving (Eq, Read, Show, Generic)
+
 instance (Monad m, MonadDecrypt (ConfCryptM m key) key) => Command ReadConfCrypt (ConfCryptM m key) where
-    evaluate _ = do
+    evaluate (ReadConfCrypt template) = do
         (ccFile, ctx) <- ask
         let params = parameters ccFile
-        transformed <- mapM (\p -> decryptedParam  p <$> decryptValue ctx (paramValue p)) params
-        processReadLines transformed ccFile
+        case template of
+            Nothing -> do
+                transformed <- mapM (\p -> decryptedParam p <$> decryptValue ctx (paramValue p)) params
+                processReadLines transformed ccFile
+            Just tpl -> do
+                params' <- sequence $ decryptParam ctx <$> params
+                pure $ renderTemplate tpl <$> params'
+
         where
+            decryptParam ctx (Parameter paramName value paramType) = do
+                value' <- decryptValue ctx value
+                pure $ Parameter {paramName, paramValue = value', paramType}
```

Comment:
There is a special syntax for creating a copy of a record with one (or more) fields changed: `a { b = c }`.
So this function could look like this:

```
decryptParam ctx p = do
    value' <- decryptValue ctx $ paramValue p
    pure p { paramValue = value' }
```